Snakefile

```diff
@@ -469,6 +472,71 @@ rule fasta_headers:
             else print; }}' \
             {output.masked_consensus} > temp_{wildcards.sample}.fasta
         mv temp_{wildcards.sample}.fasta {output.masked_consensus}
+
+        """
+
+rule metadata_to_json:
+    input:
+        all_r1 = lambda wildcards: mapped[wildcards.sample][0],
+        all_r2 = lambda wildcards: mapped[wildcards.sample][1]
+    output:
+        metadata = "consensus_genomes/{reference}/{sample}.metadata.json"
+    shell:
+        """
+        python scripts/metadata_to_json.py "{input.all_r1}" "{input.all_r2}" > {output.metadata}
+        """
+
+rule masked_consensus_to_json:
+    input:
+        masked_consensus = rules.fasta_headers.output.masked_consensus
+    output:
+        masked_consensus = "consensus_genomes/{reference}/{sample}.masked_consensus.json"
+    shell:
+        """
+        python scripts/fasta_to_json.py {input.masked_consensus} > {output.masked_consensus}
+        """
+
+rule summary_stats_to_json:
+    input:
+        bam_coverage = rules.coverage_summary.output.coverage,
```

Comment:
bamstats summary info should be coming from `rules.bamstats.output.bamstats_file`